doc/host_function.md

```diff
@@ -0,0 +1,226 @@
+    addHostGlobal("global_i64", std::make_unique<Runtime::Instance::GlobalInstance>(ValType::I64, ValMut::Const, uint64_t(666)));
+    addHostGlobal("global_f32", std::make_unique<Runtime::Instance::GlobalInstance>(ValType::F32, ValMut::Const, float(666)));
+    addHostGlobal("global_f64", std::make_unique<Runtime::Instance::GlobalInstance>(ValType::F64, ValMut::Const, double(666)));
+  }
+  virtual ~TestModule() = default;
+
+private:
+  std::vector<uint8_t> &Data;
+};
+
+} // namespace Host
+} // namespace SSVM
+```
+
+`Host module` supplies `getFuncs()`, `getTables()`, `getMems()`, and `getGlobals()` to search registered instances by unique exporting name. For more details, APIs can be found in `include/runtime/importobj.h`.
+
+### Register Host Modules to SSVM
+
+Users can register host modules via `SSVM::VM::registerModule()` API.
+
+```cpp
+#include "vm/configure.h"
+#include "vm/vm.h"
+#include <vector>
+
+SSVM::VM::Configure Conf;
+SSVM::VM::VM VM(Conf);
+std::vector<uint8_t> Data;
+SSVM::Host::TestModule TestMod(Data);
+VM.registerModule(TestMod);
+```
+
+### Link Libraries And Include Directories in CMakeFile
+
+For finding headers from SSVM include directories and linking static libraries, some settings are necessary for CMakeFile:
+
+```
+add_library(ssvmHostModuleTest  # Static library name of host modules
+  test.cpp  # Path to host modules cpp files
+)
+
+target_include_directories(ssvmHostModuleTest
+  PUBLIC
+  ${Boost_INCLUDE_DIR}
+  ${PROJECT_SOURCE_DIR}/include
+  ${PROJECT_SOURCE_DIR}/thirdparty
+)
+```
+
+## Implementation of Host Function Body
+
+There are some tips about implementing host function bodies.
+
+### Checking Memory Instance When Using
+
+Host function can access WASM memory, which passed as `MemoryInstance *` argument. When a [function call occurs](https://webassembly.github.io/spec/core/exec/instructions.html#function-calls), a frame with module which the called function belonging to will be pushed onto the `stack`. In the `host function` case, the `memory` instance of the module of the top frame on the `stack` will be passed as the `host function` body's argument. But there can be no `memory` instance in a WASM module. Therefore, users should check if the memory instance pointer is a `nullptr` or not when accessing.
+
+### Returning Expectation
+
+From our mechanism, `Expect<T>` declared in `include/common/errcode.h` is used as the result type of function body. In `Expect<void>` case, `return {};` is needed for an expected situation. In other cases, `return Value;` is needed, where `Value` is a variable of type `T`. If an unexpected situation occurs, users can call `return Unexpect(Code);` to return an error, which `Code` is an element of enumeration `ErrCode`.
+
+### Forcing Termination
+
+SSVM provides a method for terminating WASM execution in host functions. Return `ErrCode::Terminated` can force terminate the execution and the execution result will be set as successful.
```

Comment:
SSVM provides a method for terminating WASM execution in host functions. Developers can return `ErrCode::Terminated` to trigger the forcing termination of the current execution and the execution result will be set as successful.